components/chat_widget/src/components/ocs-chat/ocs-chat.tsx

```diff
@@ -295,13 +318,79 @@ export class OcsChat {
     }
   }
 
+  private async uploadFiles(): Promise<number[]> {
+    if (this.selectedFiles.length === 0 || !this.sessionId) {
+      return [];
+    }
+
+    this.isUploadingFiles = true;
+    const uploadedIds: number[] = [];
+
+    try {
+      const formData = new FormData();
+
+      // Add all files to form data
+      for (const selectedFile of this.selectedFiles) {
+        if (!selectedFile.error && !selectedFile.uploaded) {
+          formData.append('files', selectedFile.file);
+        } else if (selectedFile.uploaded) {
+          uploadedIds.push(selectedFile.uploaded.id);
+        }
+      }
+
+      // Add user ID and name to the form data
+      const userId = this.getOrGenerateUserId();
+      formData.append('participant_remote_id', userId);
+      if (this.userName) {
+        formData.append('participant_name', this.userName);
+      }
+
+      // Only upload if there are new files
+      if (formData.has('files')) {
+        const response = await fetch(`${this.getApiBaseUrl()}/api/chat/${this.sessionId}/upload/`, {
+          method: 'POST',
+          body: formData,
+        });
+
+        if (!response.ok) {
+          const errorData = await response.json();
+          throw new Error(errorData.error || 'Failed to upload files');
+        }
+
+        const data = await response.json();
+
+        // Update selected files with upload results
+        let fileIndex = 0;
+        this.selectedFiles = this.selectedFiles.map(sf => {
+          if (!sf.error && !sf.uploaded) {
+            return { ...sf, uploaded: data.files[fileIndex++] };
+          }
+          return sf;
+        });
+        uploadedIds.push(...data.files.map((f: UploadedFile) => f.id));
+      }
+
+      return uploadedIds;
+    } catch (error) {
+      this.error = error instanceof Error ? error.message : 'Failed to upload files';
+      throw error;
+    } finally {
+      this.isUploadingFiles = false;
+    }
+  }
+
   private async sendMessage(message: string): Promise<void> {
     if (!this.sessionId || !message.trim()) return;
 
     // Hide starter questions on any user interaction
     this.showStarterQuestions = false;
 
     try {
+      let attachmentIds: number[] = [];
```

Comment:
I'm curious why you decided to do file uploading and message submission as two separate steps?

Reply:
From my perspective I'm fine with having them be separate for two reasons:
1. We could do the upload as soon as the file is added instead of waiting for the user to click 'send'. This will reduce the latency. (I'm not proposing we do this as part of this PR).
2. If we move to websockets, we'd need to upload the files separately anyway.